saexec/saexec_test.go

```diff
@@ -374,74 +374,83 @@ func TestGasAccounting(t *testing.T) {
 		wantPriceAfter  gas.Price
 	}{
 		{
-			target:          5 * gasPerTx,
+			// Initially set the gasTarget for the next block.
+			blockTime:       0,
+			numTxs:          0,
+			targetAfter:     5 * gasPerTx,
+			wantExecutedBy:  at(0, 0, 10*gasPerTx),
+			wantExcessAfter: 0,
+			wantPriceAfter:  1,
+		},
+		{
 			blockTime:       2,
 			numTxs:          3,
+			targetAfter:     5 * gasPerTx,
 			wantExecutedBy:  at(2, 3, 10*gasPerTx),
 			wantExcessAfter: 3 * gasPerTx / 2,
 			wantPriceAfter:  1, // Excess isn't high enough so price is effectively e^0
 		},
 		{
-			target:          5 * gasPerTx,
 			blockTime:       3, // fast-forward
 			numTxs:          12,
+			targetAfter:     5 * gasPerTx,
 			wantExecutedBy:  at(4, 2, 10*gasPerTx),
 			wantExcessAfter: 12 * gasPerTx / 2,
 			wantPriceAfter:  1,
 		},
 		{
-			target:          5 * gasPerTx,
 			blockTime:       4, // no fast-forward so starts at last execution time
 			numTxs:          20,
+			targetAfter:     5 * gasPerTx,
 			wantExecutedBy:  at(6, 2, 10*gasPerTx),
 			wantExcessAfter: (12 + 20) * gasPerTx / 2,
 			wantPriceAfter:  1,
 		},
 		{
-			target:          5 * gasPerTx,
 			blockTime:       7, // fast-forward equivalent of 8 txs
 			numTxs:          16,
-			wantExecutedBy:  at(8, 6, 10*gasPerTx),
-			wantExcessAfter: (12 + 20 - 8 + 16) * gasPerTx / 2,
+			targetAfter:     10 * gasPerTx,           // double gas/block --> halve ticking rate
+			wantExecutedBy:  at(8, 6*2, 20*gasPerTx), // ending point scales
+			wantExcessAfter: 2 * (12 + 20 - 8 + 16) * gasPerTx / 2,
```

Comment:
```suggestion
			// Doubling the target scales both the ending time and excess to compensate.
			wantExecutedBy:  at(8, 2*6, 2*10*gasPerTx),
			wantExcessAfter: 2 * (12 + 20 - 8 + 16) * gasPerTx / 2,
```

I know that the `// starting point scales` comment was originally mine and it inspired the equivalent one here, but the fact that I as the original author really had to think meant that it was inadequate 🤦